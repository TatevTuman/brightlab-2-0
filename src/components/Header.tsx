import { Link } from 'gatsby'
import React, { memo } from 'react'
import { useSiteMetadata } from '~hooks'
import * as Icon from '~svg'

interface HeaderProps {}

const Header: React.FC<HeaderProps> = props => {
  const {} = props

  const { navigation } = useSiteMetadata()

  return (
    <header className="bg-green-600 py-16">
      <div className="container">
        <nav className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8" aria-label="Top">
          <div className="w-full py-6 flex items-center justify-between border-b border-green-500 lg:border-none">
            <div className="flex items-center">
              <Icon.Logo className="text-white -mt-9 mr-30" />
              <div className="space-x-32 lg:block">
                {navigation.map(link => (
                  <Link
                    key={link.label}
                    to={link.path}
                    className="text-base font-medium text-white hover:text-green-200"
                  >
                    {link.label}
                  </Link>
                ))}
              </div>
            </div>
            <div className="ml-10 space-x-4">
              <button className="btn btn-green">Sign In</button>
              <button className="btn btn-green-white">Sign up</button>
            </div>
          </div>
        </nav>
      </div>
    </header>
  )
}

export default memo(Header)
