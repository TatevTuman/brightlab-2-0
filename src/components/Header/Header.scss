.header {
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;
  background: #fff;

  &-content {
    display: flex;
    justify-content: flex-end;
  }
}
