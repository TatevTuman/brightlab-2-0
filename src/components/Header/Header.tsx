import React, { memo } from 'react'
import { Container } from '@components'
import { Link } from '@elements'
import { useSiteMetadata, useWindowSize } from '@hooks'
import styles from './Header.module.scss'
import LogoImage from '@images/logo.svg'

const HomeLogo = () => {
  const { toRender } = useWindowSize(['desktop'])
  if (!toRender) return null

  const logoStyles = {
    position: 'absolute' as const,
    top: 0,
    right: 0
  }

  return <LogoImage width={100} height={100} style={logoStyles} />
}

export interface HeaderProps {}

const Header: React.FC<HeaderProps> = props => {
  const { navigation } = useSiteMetadata()
  return (
    <div className={styles.header}>
      <Container>
        <HomeLogo />
        <nav role={'navigation'} data-direction="horizontal">
          <ul>
            {navigation.map(item => {
              const { path, label } = item

              return (
                <li key={path}>
                  <Link to={path}>{label}</Link>
                </li>
              )
            })}
          </ul>
        </nav>
      </Container>
    </div>
  )
}

export default memo(Header)
