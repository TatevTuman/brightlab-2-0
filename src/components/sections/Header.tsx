import { Link } from 'gatsby'
import React, { memo } from 'react'
import { useSiteMetadata } from '~hooks'
import { Button } from '~ui'
import * as Icon from '~svg'

export interface HeaderProps {
}

const Header: React.FC<HeaderProps> = props => {
  const {} = props

  const { navigation } = useSiteMetadata()

  return (
    <header className="pt-24">
      <div className="container">
        <nav className="mx-auto px-4 sm:px-6 lg:px-8" aria-label="Top">
          <div className="w-full py-6 flex items-center justify-between">
            <div className="flex items-center">
              <Icon.Logo className="w-143 h-30 fill-black cursor-pointer"/>
            </div>
            <div className="text-26">
              <div className="flex space-x-62">
                {navigation.map(link => (
                  <Link key={link.label} to={link.path} className="text-26">
                    {link.label}
                  </Link>
                ))}

                {/*Refactor ===========>*/}

                <div className="space-x-20">
                  <button className="text-20 text-gray hover:text-black">RU</button>
                  <button className="text-20">ENG</button>
                </div>

              {/* =============/ */}

              </div>
            </div>
          </div>
        </nav>
      </div>
    </header>
  )
}

export default memo(Header)
