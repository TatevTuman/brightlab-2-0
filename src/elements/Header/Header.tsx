import React, { memo } from 'react'
import { Navigation, Container } from '@elements'
import { useSiteMetadata } from '@hooks'
import './Header.scss'

const HomeLogo = () => {
  return null
}

export interface HeaderProps {}

const Header: React.FC<HeaderProps> = props => {
  const { navigation } = useSiteMetadata()

  return (
    <div className={'header'}>
      <Container>
        <HomeLogo />
        <Navigation navigation={navigation} activeMatch />
      </Container>
    </div>
  )
}

export default memo(Header)
