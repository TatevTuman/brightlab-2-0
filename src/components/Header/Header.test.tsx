import React from 'react'
import { render, cleanup, waitFor } from '@testing-library/react'
import Header from './Header'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('Header', () => {
  it('renders correctly', async () => {
    const { container } = render(<Header />)
    const awaitedContainer = await waitFor(() => container)

    expect(awaitedContainer).toMatchSnapshot()
  })
})
