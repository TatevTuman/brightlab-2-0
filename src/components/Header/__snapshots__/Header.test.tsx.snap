// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="container"
    >
      <test-file-stub
        height="100"
        style="position: absolute; top: 0px; right: 0px;"
        width="100"
      />
      <nav
        data-direction="horizontal"
        role="navigation"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/typography"
            >
              Typography
            </a>
          </li>
          <li>
            <a
              href="/sign-in"
            >
              Sign in
            </a>
          </li>
          <li>
            <a
              href="/sign-up"
            >
              Sign up
            </a>
          </li>
          <li>
            <a
              href="/dynamic"
            >
              Dynamic
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
